addons/sale/models/sale.py

```diff
@@ -115,6 +115,11 @@ def _compute_tax_id(self):
         for order in self:
             order.order_line._compute_tax_id()
 
+    @api.depends('pricelist_id', 'date_order', 'company_id')
+    def _compute_rate(self):
+        for order in self:
+            order.currency_rate = order.currency_id._get_rates(order.company_id, order.date_order).get(order.currency_id.id, 1.0)
```

Comment:
There is a pretty common use case where the rate is agreed between parts.

**Example:**

- In México the rate in a sale order is agreed at the date of the closing date.
- In Costa Rica the rate is at the moment of the payment.
- In Canada/EEUU relationships the Rate can be set once per year instead per transactions.
- In overinflationary countries the rate can be set at the moment of payment.

If you make this field Record = True and make possible to use such field able to be set (with an inverse) this will help A LOT, and almost effortless.

**Feature:**

By default this logic is perfect, but order by order this can be set manually or using any other logic with a simple write.

The only thing you need to do is allow the readonly=False and set the inverse to allow securelly write it.

Thanks in advance for hear me. cc @odony @qdp-odoo

Reply:
cc @jem-odoo

Reply:
This commit aims to improve the performance, and unfortunately not changing the existing feature. Since we are near the release what you asked can not be done, but I transfert your request on the Product Owner Team ;)